docs/spec.md

```diff
@@ -170,6 +174,82 @@ Informs the client that channels are no longer available.
 }
 ```
 
+### Client Advertise
+
+- Informs the server about available client channels. Note that the client is only allowed to advertise channels if the server previously declared that it has the `clientPublish` [capability](#server-info).
+
+#### Fields
+
+- `op`: string `"advertise"`
+- `channels`: array of:
+  - `id`: number chosen by the client. The client may reuse ids that have previously been unadvertised.
+  - `topic`: string
+  - `encoding`: string, must be `"json"`
+  - `schemaName`: string
+
+#### Example
+
+```json
+{
+  "op": "advertise",
+  "channels": [
+    {
+      "id": 1,
+      "topic": "foo",
+      "encoding": "json",
+      "schemaName": "ExampleMsg"
+    }
+  ]
+}
+```
+
+### Client Unadvertise
+
+- Informs the server that client channels are no longer available. Note that the client is only allowed to unadvertise channels if the server previously declared that it has the `clientPublish` [capability](#server-info).
+
+#### Fields
+
+- `op`: string `"unadvertise"`
+- `channelIds`: array of number, corresponding to previous [Client Advertise](#client-advertise)
+
+#### Example
+
+```json
+{
+  "op": "unadvertise",
+  "channelIds": [1, 2]
+}
+```
+
+### Client Message Data
```

Comment:
Nit: maybe rename this to "client publish" to match the `op`? Or rename the `op` to `"msg"` or similar?